### ISSUE TITLE
Submission: Cynogen

### ISSUE THREAD

**Issue by sugatobagchi** [gdgcloudkol/hackday#39]
### Team Name

Cynogen

### Team members

Member # 1
Name: Sugato Bagchi
Email: sugato.bagchi.of@gmail.com
Twitter handle (hyperlinked): [sugato_bagchi](https://twitter.com/sugato_bagchi)

Member # 2
Name: Subhanjan Dutta
Twitter handle (hyperlinked): [subhanjan_dutta](https://twitter.com/subhanjan_dutta)

Member # 3
Name: Anusha Bera
Twitter handle (hyperlinked): [anusha_bera](https://twitter.com/anusha_bera)

### Project Name

StockCastery

### Contact Details (Leader)

sugato.bagchi.of@gmail.com

### Project Track

Fin-Tech

### Link to project GitHub public repo

https://github.com/ethan-x11/StockCastery

### Link to project website

https://stockcastery-sugatobagchi-dsecacin.vercel.app/

### Link to demo video

https://www.youtube.com/watch?v=6Zo6Eo1GV_Y

### Inspiration

Web3 is currently a hot topic in the market. Even more important is how stable crypto currencies are and what their future price will be.

With a dash of FinTech, Machine Learning, and web3, we created the project StockCastery, which is a one-stop place for all your crypto watch, tracking, and prediction needs.

We used React to build the frontend, Material UI for style, Chart JS to construct the graph, Axios to fetch data from API, CoinGeko API for data, Tensorflow Framework & Keras Model to build the prediction model, Ski Kit Learn to analyse future data, and Vercel to host the website. And I'm very happy to share that we now have more than 70% accuracy across practically all crypto shares prediction, and we plan to strengthen this model in the next days.

Our approach is also open source, and our website is free. We do not require you to login or register. We are not requesting information about your wallet. The website is entirely free, responsive, and simple to use.

### What it does

- StockCastery is a live and trending cryptocurrency tracking service.

- It displays the current price of your favourite cryptocurrency in real time.
-  It also forecasts your favourite cryptocurrency shares based on real-time data.
- Our algorithm presently has greater than 70% accuracy, and we also consider other elements such as a sudden decline, unexpected strong increase, and so on when estimating pricing.
- It includes graphs and details for each day data.
- It shows market volume and crypto details as well.

### Challenges you ran into

- The first and most significant problem we encountered was developing an accurate and fast prediction model. The backend team worked tirelessly to develop a good prediction model and incorporate it into the site, which takes very little time to predict data after analysis.

- The second difficulty was to create a responsive site while keeping all of the features we considered. The frontend team reviewed numerous materials and created a fully responsive site for this.

- The third issue we encountered was backend deployment. Flask was used to build the backend. Fortunately, we discovered a site and created a sort of API. You can now send a post request to our backend link and obtain a prediction in a matter of seconds.

### Anything else?

Prospects for the Future:

- Improving the prediction model by incorporating more data.
- We intend to include a news section in the app, where you can access trending headlines as well as information about a specific cryptocurrency.
- Putting in place a crypto resource database. After clicking on the link, anyone can learn about cryptocurrencies.

![Screenshot (238)](https://user-images.githubusercontent.com/68023376/185971204-99e2fffe-b902-413b-9d6f-22f455e8a10b.png)

![Screenshot (233)](https://user-images.githubusercontent.com/68023376/185970911-a5228fb0-d58f-4a62-a6d7-ffaa5ecb44ca.png)

![Screenshot (236)](https://user-images.githubusercontent.com/68023376/185970907-2e764005-2e06-4518-9f5d-de7bd75dff35.png)

![Screenshot (237)](https://user-images.githubusercontent.com/68023376/185970902-a2fa6525-93dc-4570-a83d-eaf76801ec5c.png)

**Comment by xprilion** [comment]
Kindly provide a deployment for this commit - https://github.com/ethan-x11/StockCastery/commit/a439d2ceeedaaafcd6fc4654c76775a49bb42ba9

**Comment by sugatobagchi** [comment]
> Kindly provide a deployment for this commit - [ethan-x11/StockCastery@a439d2c](https://github.com/ethan-x11/StockCastery/commit/a439d2ceeedaaafcd6fc4654c76775a49bb42ba9)

@xprilion You can now visit the same link i.e. [here](https://stockcastery-sugatobagchi-dsecacin.vercel.app/). I have reverted back the web-app to the same desired commit which you have stated.

Let me know if you need anything else!
